src/postgres-util/src/replication.rs

```diff
@@ -141,3 +142,37 @@ pub async fn drop_replication_slots(
     }
     Ok(())
 }
+
+/// Fetches the active replication timeline ID from an upstream Postgres source.
+///
+/// Will start a replication connection to allow using the IDENTIFY_SYSTEM call
+/// for fetching the timeline ID.
+pub async fn get_timeline_id(
+    ssh_tunnel_manager: &SshTunnelManager,
+    config: &Config,
+) -> Result<u32, PostgresError> {
+    let replication_client = config.connect_replication(ssh_tunnel_manager).await?;
+
+    get_timeline_id_with_client(&replication_client).await
+}
+
+/// Fetches the active replication timeline ID from an upstream Postgres source.
+pub async fn get_timeline_id_with_client(
+    replication_client: &Client,
+) -> Result<u32, PostgresError> {
+    if let SimpleQueryMessage::Row(r) =
+        &replication_client.simple_query("IDENTIFY_SYSTEM").await?[0]
+    {
+        match r.get("timeline").expect("Returns one row").parse::<i8>() {
+            Ok(timeline_id) => Ok(timeline_id as u32),
```

Comment:
We don't allow `as` functions––you can look in `mz_ore` for these types of fallible conversions.